File: dan/Problems/Easy/LinkedList/203. Remove Linked List Elements/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if not head:
            return
        while head and head.val == val:
            head = head.next
        
        if head:
            pre, cur = head, head.next
            while cur:
                while cur and cur.val == val:
                    pre.next, cur = cur.next, cur.next
                if cur:
                    pre, cur = cur, cur.next
        return head

File: dan/Problems/Easy/LinkedList/203. Remove Linked List Elements/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null)
            return head;
            
        ListNode cur = head;
        while(cur != null && cur.next != null){
            while(cur.next != null && cur.next.val == val)
                cur.next = cur.next.next;
            cur = cur.next;  
        }
        return head.val == val? head.next : head;
    }
}
